File: problems/finding_3-digit_even_numbers/solution.py
class Solution:
    def findEvenNumbers(self, digits: List[int]) -> List[int]:
        freq = Counter(digits)
        ans = []
        
        def dfs(length: int, num: int) -> None:
            if length == 0:
                if num % 2 == 0:
                    ans.append(num)
                return
            
            for i in range(10):
                if freq[i] == 0 or not freq[i]:
                    continue
                
                if length == 3 and i == 0:
                    continue
                    
                freq[i] -= 1
                dfs(length - 1, 10 * num + i)
                freq[i] += 1
        dfs(3, 0)
        
        return ans

File: problems/finding_3-digit_even_numbers/solution.cs
public class Solution {
    public int[] FindEvenNumbers(int[] digits) {
        int[] digitCount = new int[10];
        foreach(int d in digits)
            digitCount[d]++;
        List<int> output = new();
        for(int i = 100; i < 999; i += 2) {
            byte d1 = (byte)(i % 10), 
                d2 = (byte)((i / 10) % 10),
                d3 = (byte)((i / 100) % 10);
            digitCount[d1]--;
            digitCount[d2]--;
            digitCount[d3]--;

            bool flag = true;
            if(digitCount[d1] < 0 || 
               digitCount[d2] < 0 ||
               digitCount[d3] < 0) {
                   flag = false;
            }

            digitCount[d1]++;
            digitCount[d2]++;
            digitCount[d3]++;

            if(flag)
                output.Add(i);
        }
        return output.ToArray();
    }
}